### ISSUE TITLE
Focus is not restored when close a Callout controlled by hidden

### ISSUE THREAD

**Issue by Deerhound579** [microsoft/fluentui#19179]
### Environment Information

- **Package version(s)**: 8.25.0
- **Browser and OS versions**: Chrome 91.0.4472.164 (Official Build) (64-bit) + Windows 10

### Describe the issue:

When using keyboard, ESC closes the callout, but the focus is not restored to the previous focused element. This happens when `Callout` is controlled by `hidden`.

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/Deerhound/pen/JjNLZyq

* Use keyboard to focus the button
* Press `Enter` to open the callout
* Press `Esc` to close the callout

#### Actual behavior:

The focus is restored correctly for the first one but lost for the other.

#### Expected behavior:

Focus should be restored when controlled by `hidden`.

### Documentation describing expected behavior

From [wai-aria-practices-1.1](https://www.w3.org/TR/wai-aria-practices-1.1/#menu)
![image](https://user-images.githubusercontent.com/43892874/127467087-e5eed760-c938-4b62-ae89-1029cc69563b.png)

**Comment by ling1726** [comment]
@Deerhound579  Thanks for the great issue and repro!
